floss/language_identifier.py

```diff
@@ -0,0 +1,70 @@
+# Copyright (C) 2023 Mandiant, Inc. All Rights Reserved.
+import pefile
+
+import floss.logging_
+from floss.logging_ import DebugLevel
+
+logger = floss.logging_.getLogger("floss")
+
+
+def is_go_bin(sample: str) -> bool:
+    """
+    Check if the binary given is compiled with Go compiler or not
+    it checks the magic header of the pclntab structure -pcHeader-
+    the magic values varies through the version
+    reference:
+    https://github.com/0xjiayu/go_parser/blob/865359c297257e00165beb1683ef6a679edc2c7f/pclntbl.py#L46
+    """
+    try:
+        pe = pefile.PE(sample)
+    except:
+        logger.debug("NOT valid PE header")
+        return False
```

Comment:
see the tests issue, we should at least catch FileNotFound here specifically and maybe also pefile errors.